tests/PHPStan/Analyser/data/value-of-enum.php

```diff
@@ -0,0 +1,36 @@
+<?php // lint >= 8.1
+
+declare(strict_types=1);
+
+namespace ValueOfEnum;
+
+use function PHPStan\Testing\assertType;
+
+enum Country: string
+{
+	case NL = 'The Netherlands';
+	case US = 'United States';
+}
+
+class Foo {
+	/**
+	 * @return value-of<Country>
+	 */
+	function us()
+	{
+		return Country::US;
+	}
+
+	/**
+	 * @param value-of<Country> $countryName
+	 */
+	function hello($countryName)
+	{
+		assertType('string', $countryName);
```

Comment:
This isn't good enough, right? Should be `'The Netherlands'|'United States'` really.

Reply:
It's also why that bug isn't report in CallMethodsRuleTest.

Reply:
fixed it. good catch.